### PR TITLE
add a compact widget showing only health, armor and ammo information

### LINKED ISSUE #1396: [Feature] Compact "essentials" for WOOFHUD
How feasible would it be to add an optional display of just the essential stats (Health, Armor, Ammo) in a single line to the WOOFHUD lump (similar to the K/I/S stats)?

Explanation:
Instead of using three separate lines, it would look like this (example):
**HEL 100   ARM 100   AMM 50/50** 
(No progress bars, just the numbers, i.e. _hud_draw_bargraphs_ would have to be set to 0 automatically)

Reasoning:
For absolute minimalists (like me) who want to save as much screen space as possible while still being able to check the most vital stats. Also maybe for those who consider the bars kinda redundant. [Differing from my previous suggestion](https://github.com/fabiangreffrath/woof/issues/1390), this approach would not mix different HUD styles, just reorganize existing ones a bit. 

Possible realization:
- Either introduce a CVAR which toggles between the old look and the single-line mode (setting _hud_draw_bargraphs_ to 0 could e.g. automatically align the three widgets in a single line automatically, showing as many as you have activated)
- Add a new (optional) identifier to WOOFHUD, something like "vitals", "essentials" or "compact"

### PR DESCRIPTION
**Pull request by fabiangreffrath**
I couldn't get it done with the existing widgets, so I created a new one which displays only the most basic information (health, armor, ammo) in one line. Fixes #1396

### CODE REVIEW

File: src/hu_stuff.c [hunk 1]
```diff
@@ -857,32 +902,76 @@ static void HU_widget_build_armor (void)
...
+  const crange_idx_e cr = CRByArmor();
+  w_armor.cr = colrngs[cr];
+
+  // transfer the init string to the widget
+  HUlib_add_string_to_cur_line(&w_armor, hud_armorstr);
+}
+
+static void HU_widget_build_compact (void)
+{
+  char hud_compactstr[HU_MAXLINELENGTH];
```
Comment (fabiangreffrath) [inline src/hu_stuff.c:914]: It never happens that two widget strings are built at the same time. At this point, would it make sense to use a common static global string buffer for all widgets?

**Comment by fabiangreffrath** [comment]
I have to say that I could get used to it. it's really good!

woof0000

File: src/hu_stuff.c (hunk 1 above)
Comment (rfomin) [reply src/hu_stuff.c:914]: Sure, if it's more convenient. We need to make sure that this global buffer always has `0` at the end, right?

**Review by rfomin — APPROVED** [review]

**Comment by NightFright2k19** [comment]
Looks really good! Will be perfect for minimal HUD mode. A pity I gotta wait for another 14d before I can check this out.

Thanks a lot for making it happen!

**Pull request merged**